tokio/src/task/join_set.rs

```diff
@@ -220,35 +229,44 @@ impl<T: 'static> JoinSet<T> {
     ///
     /// # Cancel Safety
     ///
-    /// This method is cancel safe. If `join_one_with_id` is used as the event in a `tokio::select!`
+    /// This method is cancel safe. If `join_next_with_id` is used as the event in a `tokio::select!`
     /// statement and some other branch completes first, it is guaranteed that no tasks were
     /// removed from this `JoinSet`.
     ///
     /// [task ID]: crate::task::Id
     /// [`JoinError::id`]: fn@crate::task::JoinError::id
+    #[doc(alias = "join_one_with_id")]
+    pub async fn join_next_with_id(&mut self) -> Option<Result<(Id, T), JoinError>> {
+        crate::future::poll_fn(|cx| self.poll_join_next(cx)).await
+    }
+
+    #[doc(hidden)]
+    #[deprecated(
+        note = "This method is deprecated and will be removed. Use `JoinSet::join_next_with_id` instead."
```

Comment:
nits: normally these deprecation notices start with a lowercase letter, because the compiler error is formatted like `warning: use of deprecated API: <text of deprecation note>`. also, i think this could be simplified to just
```suggestion
        note = "renamed to `JoinSet::join_next_with_id`"
```